client/js/lounge.js

```diff
@@ -1142,13 +1207,34 @@ $(function() {
 			var chan = $(this);
 			if (chan.find(".messages .msg:not(.unread-marker)").slice(0, -100).remove().length) {
 				chan.find(".show-more").addClass("show");
+
```

Comment:
I spent a long time trying to think of a nicer way to do this.

You could obviously select the first child that isn't a date-change-marker, but I don't think there's a way to then delete all except 1 of the date-change-markers before it. And even if you did, jquery would still be doing the same amount of work, so ... meh.

Having said that, I think that this code could be run on initial page load and on each message, rather than on the set interval. We know that the only time this is going to be necessary is if we've had a new message received (or on initial load), don't we?